lib/dynamodb.js

```diff
@@ -1228,176 +1341,163 @@ DynamoDB.prototype.range = function (model, id, start, end, callback) {
-        }
-        break;
-      case 'lt':
-        if (useIndex) {
-          addKeyCondition(key, "LT", conditionValue.lt);
-        } else {
-          prepFilterExpression();
-          params.FilterExpression += "(#" + key + " < " + getExpressionValueToken(conditionValue.lt) + ")";
-        }
-        break;
-      case 'lte':
-        if (useIndex) {
-          addKeyCondition(key, "LE", conditionValue.lte);
-        } else {
-          prepFilterExpression();
-          params.FilterExpression += "(#" + key + " <= " + getExpressionValueToken(conditionValue.lte) + ")";
-        }
-        break;
-      case 'between':
-        if (useIndex) {
-          addKeyCondition(key, "BETWEEN", conditionValue.between[0], conditionValue.between[1]);
-        } else {
-          prepFilterExpression();
-          params.FilterExpression += "(#" + key + " BETWEEN " + getExpressionValueToken(conditionValue.between[0]) + " AND " + getExpressionValueToken(conditionValue.between[1]) + ")";
-        }
-        break;
-      case 'nin':
-        prepFilterExpression();
-        params.FilterExpression += "NOT " + "(#" + key + " IN (" + getExpressionValueToken(conditionValue.nin) + "))"
-        break;
-      case 'near':
-        throw new Error('near is not supported');
-        break;
-      case 'neq':
-        prepFilterExpression();
-        params.FilterExpression += "(#" + key + " <> " + getExpressionValueToken(conditionValue.neq) + ")";
-        break;
-      case 'like':
-        prepFilterExpression();
-        params.FilterExpression += "(contains(#" + key + "," + getExpressionValueToken(conditionValue.like) + "))";
-        break;
-      case 'nlike':
-        prepFilterExpression();
-        params.FilterExpression += "(NOT contains(#" + key + "," + getExpressionValueToken(conditionValue.nlike) + "))";
-        break;
-      case 'or':
-      case 'and':
-        throw new Error('and and or conditions are not supported');
+  let analyzeAndOr = (params, conditionValue, operator) => {
+    let returnExpressions = [];
+    conditionValue.forEach((obj) => {
+      let key = Object.keys(obj)[0];
+      let value = obj[key];
+      if (key === 'and') {
+        returnExpressions.push(analyzeAndOr(params, value, ' AND '));
+      } else if (key === 'or') {
+        returnExpressions.push(analyzeAndOr(params, value, ' OR '));
+      } else {
+        returnExpressions.push(addExpressionToParam(params, key, value));
       }
     });
-  } else {
-    //Simple case -- do equivalency in query/scan notation.
-    if (isIndexed(key, "eq", conditionValue, index)) {
-      addKeyCondition(key, "EQ", conditionValue);
+
+    if (returnExpressions.length > 1) {
+      return '(' + returnExpressions.join(operator) + ')';
     } else {
-      prepFilterExpression();
-      params.FilterExpression += "#" + key + " = " + getExpressionValueToken(conditionValue);
+      return returnExpressions[0];
     }
   }
-};
+
+  let analyzeQueryWhere = (params, queryWhere) => {
```

Comment:
If after we split the where, we convert it from an object to an AND array, can we get rid of analyzeQueryWhere and only use analyzeAndOr?